### DISCUSSION TITLE
Passwordless Login Rate Limits On IP Instead Of Target User

### DISCUSSION THREAD

**Discussion by fischjer4** [supabase/supabase#26477] [Questions]
[I see that](https://supabase.com/docs/guides/platform/going-into-prod#rate-limiting-resource-allocation--abuse-prevention) the rate-limits for `/auth/v1/verify` are applied at the IP Address layer. Shouldn't they be applied at the _target user layer_ instead?

I want to implement passwordless login via magic links, but the brute force protection being tied to a rate limit on the ip gives me pause. Couldn't an attacker just bounce around and change their IP, make requests to `/verify` in parallel, and then brute force the OTP before the expiry window is up? I feel like the rate limit should be applied at the target user level so instead of "X requests from IP in time window Y" it is "X requests to log in as _target user_ in time window Y." 

What am I missing here?

**Comment by corepay** [comment]
Good point and to add why it should be at user level is for setups like me where all auth is directed through a backend server.  If rate limit by IP address then my hundreds to several hundred million plan users will be fighting to login with say OTP that is rate limited to my server IP and can only process one OTP every 60 seconds.

What am I missing here?
